### ISSUE TITLE
postgres time Macros don't respect brackets.

### ISSUE THREAD

**Issue by kehao95** [grafana/grafana#14440]
Please include this information:
### What Grafana version are you using?
 Grafana v5.0.4 (commit: 7dc36ae) 

### What datasource are you using?
Postgres

### What OS are you running grafana on?
Docker

### What did you do?
Using 
```sql
select 
 $__time(date_trunc('hour',from_ts)),
 ...
```

### What was the expected result?
generate SQL like
```sql
select 
  date_trunc('hour',from_ts) as time
...
```

### What happened instead?
generated syntax error sql
```sql
SELECT
  date_trunc('hour' AS "time"),
...
```

Summary
The Macros doesn't seem to work as expected when the variable is no a simple dateColumn.

Consequence
For some macros there are some workarounds . like `$__time` `$__time(date_trunc('hour',from_ts)),`  can be replaced by `date_trunc('hour',from_ts) as time`
But for ` $__timeFilter()` this will be incapable when the timestamp column is not TIMESTAMP type  Because there is not direct convention without using this macro.

**Comment by sunker** [comment]
@svenklemm is this a bug?

**Comment by kehao95** [comment]
Here are some examples. fun() can be a postgresql function.

input| expected|actually|-
-|-|-|-
$__time(**A**)|**A** AS "time"|**A** AS "time"| ==
$__time(**fun(A)**)|**fun(A)** AS "time"|**fun(A AS "time")**|SyntaxErr
$__time(**fun(A,B)**)|**fun(A,B)** AS "time"|**fun(A AS "time")**|SyntaxErr 
$__timeFilter(**A**)|extract(epoch from **A**) BETWEEN ...|extract(epoch from **A**) BETWEEN ... | ==
$__timeFilter(**fun(A)**)|extract(epoch from **fun(A)** BETWEEN ...| extract(epoch from **fun(A)** BETWEEN ...|==
$__timeFilter(**fun(A,B)**)|extract(epoch from **fun(A,B)** BETWEEN ...|extract(epoch from **fun(A)** BETWEEN ...|Wrong

**Comment by JuniperJK** [comment]
Yes this is still an issue and I would consider it a bug. (As of `Grafana v6.4.3 (3a2bfb7)`)

As @kehao95 mentioned 
`$__time(fun(A))` is being turned into `fun(A AS "time")` instead of `fun(A) AS "time"`

Please take a look.

**Comment by JuniperJK** [comment]
TL;DR: There's a workaround, but this is still a bug.

I was using $__timefilter and seeing this issue.

Example:
```
SELECT
  name,
  start_time as "time",
  duration,
FROM some_table
WHERE $__timeFilter(TO_TIMESTAMP(start_time))
```
was becoming
```
SELECT
  name,
  start_time as "time",
  duration,
FROM some_table
WHERE TO_TIMESTAMP(start_time BETWEEN '2019-12-08T21:03:18.487Z' AND '2019-12-10T21:03:18.487Z')
```

**Workaround**
Use $__timeFrom and $__timeTo directly.
```
SELECT
  name,
  start_time as "time",
  duration,
FROM some_table
WHERE TO_TIMESTAMP(start_time) BETWEEN $__timeFrom() AND $__timeTo()
```

**Comment by oscarkilhed** [comment]
This seems to be the same issue as #10380
